examples/modular-drivers/src/bme280_ulp.rs

```diff
@@ -0,0 +1,970 @@
+            .lock()
+            .unwrap()
+            .write_i2c(BME280_DEV_ADDR, &[Register::RESET.into(), BME280_RESET])?;
+        // reset time is around 2ms
+        sleep(Duration::from_millis(5));
+        Ok(())
+    }
+
+    fn configure(&mut self) -> Result<(), SensorError> {
+        let ctrl_meas = self.config.ctrl_meas_value();
+        let ctrl_hum = self.config.ctrl_hum_value();
+        let config = self.config.ctrl_config_value();
+
+        self.reset()?;
+        self.driver
+            .lock()
+            .unwrap()
+            .write_i2c(BME280_DEV_ADDR, &[Register::CTRL_HUM.into(), ctrl_hum])?;
+        self.driver
+            .lock()
+            .unwrap()
+            .write_i2c(BME280_DEV_ADDR, &[Register::CTRL_MEAS.into(), ctrl_meas])?;
+        self.driver
+            .lock()
+            .unwrap()
+            .write_i2c(BME280_DEV_ADDR, &[Register::CONFIG.into(), config])?;
+        Ok(())
+    }
+
+    fn read_calibration(&mut self) -> Result<(), SensorError> {
+        let mut data = [0_u8; BME280_CALIB_LEN];
+        self.driver.lock().unwrap().write_read_i2c(
+            BME280_DEV_ADDR,
+            &[Register::CALIB_DATA1.into()],
+            &mut data[..BME280_CALIB_PART1],
+        )?;
+        self.driver.lock().unwrap().write_read_i2c(
+            BME280_DEV_ADDR,
+            &[Register::CALIB_DATA2.into()],
+            &mut data[BME280_CALIB_PART1..],
+        )?;
+
+        self.calib = data.into();
+        Ok(())
+    }
+    fn read_status(&mut self) -> Result<Status, SensorError> {
+        let mut status = [0_u8; 1];
+        self.driver.lock().unwrap().write_read_i2c(
+            BME280_DEV_ADDR,
+            &[Register::STATUS.into()],
+            &mut status,
+        )?;
+        Ok(status[0].into())
+    }
+    fn read_raw_measurement(&mut self) -> Result<RawMeasurement, SensorError> {
+        let mut raw = [0_u8; BME280_MEAS_LEN];
+
+        if let Mode::FORCED = self.config.mode {
+            let ctrl_meas = self.config.ctrl_meas_value();
+            self.driver
+                .lock()
+                .unwrap()
+                .write_i2c(BME280_DEV_ADDR, &[Register::CTRL_MEAS.into(), ctrl_meas])?;
+
+            loop {
+                if self.read_status()?.measuring == 0 {
+                    break;
+                }
+            }
+        }
+
+        self.driver.lock().unwrap().write_read_i2c(
+            BME280_DEV_ADDR,
+            &[Register::PRES_REG.into()],
+            &mut raw,
+        )?;
+
+        let raw: RawMeasurement = raw.into();
+        Ok(raw)
+    }
+}
+
+impl Sensor for BME280 {}
+
+impl Readings for BME280 {
+    fn get_generic_readings(
+        &mut self,
+    ) -> Result<micro_rdk::common::sensor::GenericReadingsResult, SensorError> {
+        let measurement = self.read_raw_measurement()?;
+        let measurement = CompensatedMeasurement::from_raw(measurement, &mut self.calib);
+        let mut res = GenericReadingsResult::new();
```

Comment:
maybe implement `From<RawMeasurement>` for `GenericReadingsResult`?

Reply:
we need the calibration data so can't impl from